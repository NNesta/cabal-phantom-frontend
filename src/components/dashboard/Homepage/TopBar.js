import React from 'react';
import { InputGroup, InputGroupAddon, Input, Button } from 'reactstrap'
import SearchIcon from '@material-ui/icons/Search';
import HomeIcon from '@material-ui/icons/Home';
import PersonOutlineIcon from '@material-ui/icons/PersonOutline';
import '../../../assets/styles/dashboard.scss'
import IconButton from '../helpers/IconButton'
import { Link } from "react-router-dom";

const TopBar = () => {
    return (
        <div className="topbar">
            <form action="">
                <InputGroup>
                    <Input placeholder="Search" />
                    <InputGroupAddon addonType="append">
                        <Button color="success"><SearchIcon style={{ fontSize: "2rem" }} /></Button>
                    </InputGroupAddon>
                </InputGroup>
            </form>
            <div className="">
                <Link to= "/admin"><IconButton outline Icon={<HomeIcon />} name={"Home"} className="ml-5" href="/admin"/></Link>
                <IconButton color="info" className="ml-5" Icon={<PersonOutlineIcon />} name={"Profile"} />
            </div>
        </div>
    );
}

export default TopBar;
