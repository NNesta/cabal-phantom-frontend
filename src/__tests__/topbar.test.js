import TopBar from '../components/dashboard/Homepage/TopBar';
import React from 'react';
import { render, fireEvent, cleanup, waitFor } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/store';
import userEvent from '@testing-library/user-event';
import { BrowserRouter as Router, Link } from 'react-router-dom';


const TopBarComponent = () => {
	return render(
		<Provider store={store}>
				<Router>
					<TopBar />
				</Router>
		</Provider>
	);
};

describe('TopBar component', () => {
    afterEach(cleanup)

    it('Should render add user component properly', () => {
		const { asFragment } = TopBarComponent();
		expect(asFragment(<TopBar />)).toMatchSnapshot();
	});
})
