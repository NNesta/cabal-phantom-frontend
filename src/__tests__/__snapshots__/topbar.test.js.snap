// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopBar component Should render add user component properly 1`] = `
<DocumentFragment>
  <div
    class="topbar"
  >
    <form
      action=""
    >
      <div
        class="input-group"
      >
        <input
          class="form-control"
          placeholder="Search"
          type="text"
        />
        <div
          class="input-group-append"
        >
          <button
            class="btn btn-success"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              style="font-size: 2rem;"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
              />
            </svg>
          </button>
        </div>
      </div>
    </form>
    <div
      class=""
    >
      <a
        href="/"
      />
      <div
        class="button_icon"
      >
        <a
          href="/"
        />
        <a
          class="ml-5 btn btn-outline-secondary"
          href="/admin"
        >
          <div>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
              />
            </svg>
          </div>
          <div
            class="ml-2"
          >
            Home
          </div>
        </a>
      </div>
      <div
        class="button_icon"
      >
        <button
          class="ml-5 btn btn-info"
        >
          <div>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 5.9c1.16 0 2.1.94 2.1 2.1s-.94 2.1-2.1 2.1S9.9 9.16 9.9 8s.94-2.1 2.1-2.1m0 9c2.97 0 6.1 1.46 6.1 2.1v1.1H5.9V17c0-.64 3.13-2.1 6.1-2.1M12 4C9.79 4 8 5.79 8 8s1.79 4 4 4 4-1.79 4-4-1.79-4-4-4zm0 9c-2.67 0-8 1.34-8 4v3h16v-3c0-2.66-5.33-4-8-4z"
              />
            </svg>
          </div>
          <div
            class="ml-2"
          >
            Profile
          </div>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
